src/api/parser/Dockerfile

```
@@ -1,13 +1,34 @@
-FROM node:lts-alpine as base
+## Base ###########################################################################
+# Set up the base layer
+FROM node:16-alpine3.15 as base
 
-RUN apk add dumb-init
+RUN npm i -g pnpm
 
 WORKDIR /app
 
+## Dependencies ###################################################################
+# Stage for installing prod dependencies
+
+FROM base as dependencies
+
+COPY package.json ./
+
+RUN pnpm install --prod
+
+## Deploy ########################################################################
+# Stage for running our app
+FROM base as deploy
+
 COPY --chown=node:node . .
 
-RUN npm install --only=production --no-package-lock
+COPY --chown=node:node --from=dependencies /app/node_modules ./node_modules
+
+ENV PORT=10000
```

I think we can replace to `PARSER_PORT`
```suggestion
ENV PARSER_PORT=10000
```